app/routes/user/endorsements.tsx

```diff
@@ -215,49 +222,101 @@ export function EndorsementRequestCard({
 }) {
   const fetcher = useFetcher()
   const disabled = fetcher.state !== 'idle'
+  const [showMore, setShowMore] = useState(false)
+  const [isOverflowing, setIsOverflowing] = useState(true)
+  const ref = useRef<HTMLDivElement>(null)
+
+  function overflowing() {
+    if (ref.current)
+      setIsOverflowing(ref.current?.offsetWidth < ref.current?.scrollWidth)
+  }
+
+  useEffect(() => overflowing(), [])
```

Comment:
This dependency list is probably incorrect. It at least depends on `ref.current`.

Reply:
Mightn't you need to revalidate if the component changes dimensions, due to a change in the window size or device orientation. Would something like [useMesaure](https://usehooks.com/usemeasure) help here?

Reply:
It looks useful, but is it supported by typescript? I tried adding the package but had errors. I found https://usehooks-ts.com/ but that does not have a useMeasure hook. Also the page for [useMeasure](https://usehooks.com/usemeasure) is showing incorrect info, there are two issues about it on their github

Reply:
https://usehooks-ts.com/react-hook/use-element-size

Reply:
thats handy, implemented it in the latest commit